tests/test_e2e_acceptance.py

```
@@ -215,3 +215,107 @@ def test_absorb_promoted_slugs_never_missing(self, pipeline, synthetic_deep_dive
+        path the orchestration would have created, and asserts that
+        step_absorb un-stages those paths back to the original
+        ``synthetic_deep_dive`` location before returning.
+        """
+        # The qualified_files path in step_absorb stages files into a
+        # tempfile.TemporaryDirectory under layout.logs_dir.  We inject
+        # a mock that observes the staging tempdir and returns a
+        # synthesized result whose ``file`` field IS the staging path
+        # — exactly what the real workflow does.
+        captured_staged_paths: list[str] = []
+
+        def fake_run_absorb_workflow(vault_dir, *, directory, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `fake_run_absorb_workflow`

(ANN202)